src/lib/components/DeckGLMap/layers/piechart/pieChartLayer.ts

```diff
@@ -0,0 +1,96 @@
+import { CompositeLayer, COORDINATE_SYSTEM } from "@deck.gl/core";
+import { GeoJsonLayer, GeoJsonLayerProps } from "@deck.gl/layers";
+import { CompositeLayerProps } from "@deck.gl/core/lib/composite-layer";
+import { Feature } from "geojson";
+
+export interface PieChartLayerProps<D> extends CompositeLayerProps<D> {
+    pieRadius: number;
+}
+
+const defaultProps = {};
+export default class PieChartLayer extends CompositeLayer<
+    Feature,
+    PieChartLayerProps<Feature>
+> {
+    renderLayers(): GeoJsonLayer<Feature>[] {
+        const properties: GeoJsonLayerProps<Feature> = {
+            id: "pie-layer",
+            data: makePies(this.props.data, this.props.pieRadius),
+            pickable: true,
+            coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
+            getLineWidth: 1,
+            lineWidthUnits: "pixels",
+            stroked: false,
+            getLineColor: () => [0, 0, 0],
+            getFillColor: (d) => d?.properties.color,
+        };
+
+        const layer = new GeoJsonLayer<Feature>(
+            this.getSubLayerProps(properties)
+        );
+        return [layer];
+    }
+}
+
+PieChartLayer.layerName = "PieChartLayer";
+PieChartLayer.defaultProps = defaultProps;
+
+//================= Local help functions. ==================
+const EMPTY_GEOJSON = { type: "FeatureCollection", features: [] };
+
+function makePies(pies_json, R) {
+    if (pies_json.pies === undefined) {
+        return;
+    }
+
+    const json = JSON.parse(JSON.stringify(EMPTY_GEOJSON)); // deep copy
+    const colors = pies_json.colors;
+    for (const pie of pies_json.pies) {
+        //const R = pieRadius
+        const x = pie.x;
+        const y = pie.y;
+
+        makePie(json, R, x, y, pie.fractions, colors);
+    }
+    return json;
+}
+
+// Create json pie.
+function makePie(json, R, x: number, y: number, pie_fractions, colors) {
+    const dA = 10; // delta angle
+
+    // Normalize
+    let sum = 0;
+    for (const frac of pie_fractions) {
+        sum += frac;
+    }
+    const fractions = pie_fractions.map((e) => e / sum);
+
+    let start_a = -90.0;
+    for (let i = 0; i < fractions.length; i++) {
+        const end_a = start_a + fractions[i] * 360.0;
+        const coordinates: number[][] = [];
+        coordinates.push([x, y]);
+        for (let a = start_a; a < end_a; a += dA) {
+            const rad = (a * (2.0 * Math.PI)) / 360.0;
+            const xx = R * Math.cos(rad) + x;
+            const yy = R * Math.sin(rad) + y;
+            coordinates.push([xx, yy]);
+        }
+        const rad = (end_a * (2.0 * Math.PI)) / 360.0;
+        const xx = R * Math.cos(rad) + x;
+        const yy = R * Math.sin(rad) + y;
+        coordinates.push([xx, yy]);
+        coordinates.push([x, y]);
+
+        const col = colors[i % colors.length]; // wraps around if not enough colors
+        json.features.push({
```

Comment:
Better create geojson object in this function and return it instead of passing by reference and fill it in here. Javascript is garbage collected and everything passes by reference, so you will not make any copies by returning the object.